views/layouts/MainLayout.jsx

```diff
@@ -7,46 +7,81 @@ import Sidebar from '../components/Sidebar'
 import api from '../api'
 import getCounts from '../fe-utils/countUtil'
 import { fetchSubjects } from '../fe-utils/fetchUtil'
-import { AuthContext, NotificationContext } from '../contexts'
+import {
+  AuthContext,
+  ConfigurationsContext,
+  NotificationContext,
+  SidebarContext,
+} from '../contexts'
 import { headerTitle } from './helpers'
 
 const MainLayout = ({ classes, theme }) => {
+  const temporary = 'temporary'
+  const persistent = 'persistent'
+  const dashboard = 'dashboard'
+  const [configurations, setConfigurations] = useContext(ConfigurationsContext)
   const [, setNotification] = useContext(NotificationContext)
+  const [openSidebar, setOpenSidebar] = useContext(SidebarContext)
+  const [showHeader, setShowHeader] = useState(false)
   const [user, setUser] = useContext(AuthContext)
   const [users, setUsers] = useState([])
+  const [persistSidebar, setPersistSidebar] = useState('persistent')
   const { pathname } = useLocation()
   const navigate = useNavigate()
-
-  const [openDrawer, setOpenDrawer] = useState(false)
   const [sideBarState, setSideBarState] = useState({
     totalDays: 0,
     totalStudies: 0,
     totalSubjects: 0,
   })
-
-  const toggleDrawer = () => setOpenDrawer(!openDrawer)
+  const handleSidebarOpen = () => setOpenSidebar(!openSidebar)
   const fetchUsers = async () => {
     const usersList = await api.users.loadAll()
     setUsers(usersList)
   }
+  const loadAllConfigurations = async (userId) => {
+    try {
+      const configurations = await api.userConfigurations.all(userId)
+
+      setConfigurations(configurations)
+    } catch (error) {
+      setNotification({ open: true, message: error.message })
+    }
+  }
+  const handleDashboardContent = () => {
+    if (pathname.includes(dashboard)) {
+      setPersistSidebar(temporary)
+      setShowHeader(false)
+    } else {
+      setPersistSidebar(persistent)
+      setShowHeader(true)
+    }
+  }
 
   useEffect(() => {
     fetchSubjects().then((acl) => {
       setSideBarState(getCounts({ acl }))
     })
     fetchUsers()
+    loadAllConfigurations(user.uid)
   }, [])
+  useEffect(() => {
+    handleDashboardContent()
+  }, [pathname])
 
   return (
     <div className={classes.root}>
       <Header
-        handleDrawerToggle={toggleDrawer}
+        configurations={configurations}
+        onClose={handleSidebarOpen}
         title={headerTitle(pathname)}
         isAccountPage={false}
+        onShow={showHeader}
```

Comment:
the `showHeader` value is a boolean not a click handler so we don't want the `on` prefix for the prop here.

Reply:
removed
[2c26c5e](https://github.com/AMP-SCZ/dpdash/pull/352/commits/2c26c5ef82ff63420783fb443c4cd7a14df00535)